### PR TITLE
SSCS-10803 update FTA state to postpone when granted

### PR DESCRIPTION
**Pull request by pk2453**
### JIRA link (if applicable) ###
[SSCS-10803](https://tools.hmcts.net/jira/browse/SSCS-10803)

### Change description ###
Update FTA state when Postponement is Granted

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java [hunk 1]
```diff
@@ -169,7 +169,7 @@ public void givenAGrantedPostponementAndReadyToList_thenClearReviewStateAndRefer
         assertThat(response.getData().getState(), is(State.READY_TO_LIST));
         assertThat(response.getData().getSscsDocument(), is(not(empty())));
         assertThat(response.getData().getPostponementRequest().getUnprocessedPostponementRequest(), is(YesNo.NO));
-        assertThat(response.getData().getDwpState(), is(DwpState.DIRECTION_ACTION_REQUIRED.getId()));
```
Comment (dattipoeHMCTS) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java:172]: I think this needs to go back to business  + POs to confirm that we're changing legacy behaviour

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java (hunk 1 above)
Comment (pk2453) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java:172]: @dattipoeHMCTS checked with BA (Makeda) and confirmed this behaviour change.

**Review by ikirsanov — DISMISSED** [review]

**Review by TeaWithLucas — DISMISSED** [review]
All good, can you please follow the guide in the PR then delete the top bit and add the Jira ticket, it helping reviewing.
Screenshot_2022-08-16-10-13-03-12_3aea4af51f236e4932235fdada7d1643

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java [hunk 2]
```diff
@@ -76,6 +76,7 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
             grantPostponement(sscsCaseData, postponementRequest);
             setHearingDateToExcludedDate(sscsCaseData, response);
             cancelHearing(sscsCaseData);
+            sscsCaseData.setDwpState(DwpState.HEARING_POSTPONED.getId());
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:79]: Should be behind the S&L feature flag

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java (hunk 2 above)
Comment (filippo-possenti-version1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java]: Agreed. Please check revised code.

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by bensonlievoco — APPROVED** [review]

**Pull request merged**